src/stactools/datacube/stac.py

```diff
@@ -76,6 +76,32 @@ def get_time_offset_and_step(unit: str) -> Tuple[datetime, timedelta]:
     raise ValueError(f"Failed to parse time unit from '{unit}'")
 
 
+def iso_duration(td: timedelta) -> str:
+    date_intervals = [
+        ('W', 604800),  # 60 * 60 * 24 * 7
+        ('D', 86400),    # 60 * 60 * 24
+    ]
```

Comment:
Don't we need a `Y` and `M` as well?

Reply:
Makes sense, we should just be careful to document that this is an "opinionated" function and explain the reasoning why.